### PR TITLE
Add support for top-level await

### PR DESCRIPTION
**Pull request by duncup**
This package not support top level await before, but alfy support.

### CODE REVIEW

**Comment by duncup** [comment]
@sindresorhus @SamVerschueren 
Please review this.

**Comment by sindresorhus** [comment]
Can you explain your changes? Why are you now bundling the `run-node` script?

**Comment by SamVerschueren** [comment]
I don't see why these changes would make it support top level await all of a sudden? It has to do with the underlying Node version that's being used I assume?

Unless Alfred is using one of these environment variables https://github.com/sindresorhus/run-node/commit/9a69ce77e8995f53b3ba21fb8b8369e1550fa974, I don't see how these changes would actually make it work?

**Comment by duncup** [comment]
Alfy have support top level await.
https://github.com/sindresorhus/alfy/pull/109
IF i use it, alfy-test has been not work.
And i never see the code in run-node, just copy from the case use in alfy. 

@SamVerschueren 

----------

There always run it, and isee the pr for alfy how add top level await, and got this script, so i copy it. because i don't think change the run-node is a good idea.
@sindresorhus

**Comment by duncup** [comment]
Anything?

**Comment by SamVerschueren** [comment]
So the thing here is that `alfy-test` uses a different `run-node` than `alfy`. It appears that `alfy` has a custom `run-node` script that enables `ESM` right [here](https://github.com/sindresorhus/alfy/pull/109/files#diff-b4d5381c19220196a4f924339b9e5beaf8588e23b2d3c2c0d2320d1436a762e2R46).

@sindresorhus would it make sense to create a new `alfy-run-node` package or something that can be used by `alfy-test` as well? I guess that would allow us to support ESM here as well.

**Comment by sindresorhus** [comment]
I would just copy-paste it here for now. Node.js will eventually support top-level await by default. We can open an issue to revert back to `run-node` at such time.

**Comment by duncup** [comment]
Anything?

**Comment by duncup** [comment]
Did this project alive?

**Comment by SamVerschueren** [comment]
Sorry, totally missed that you updated your PR. I'll com back to you later today.

**Comment by duncup** [comment]
I know that you guys too busy, so don't be sorry.
But i really need fix this.

Anyway, thank you for you guys make this.

File: package.json [hunk 1]
```diff
@@ -20,7 +23,8 @@
 		"node": ">=10"
 	},
 	"scripts": {
-		"test": "xo && ava"
+	  "test": "xo && ava",
+	  "ava": "ava"
```
Comment (SamVerschueren) [inline package.json]:
```suggestion
	  "test": "xo && ava"
```

**Review by SamVerschueren — CHANGES_REQUESTED** [review]
One small comment.

File: package.json [hunk 2]
```diff
@@ -20,7 +23,7 @@
 		"node": ">=10"
 	},
 	"scripts": {
-		"test": "xo && ava"
+	  "test": "xo && ava"
```
Comment (SamVerschueren) [inline package.json]:
```suggestion
		"test": "xo && ava"
```

**Review by SamVerschueren — CHANGES_REQUESTED** [review]

**Comment by rchl** [comment]
This change is a bit controversial to me.

When having both `alfy` and `alfy-test` installed, the `alfy-test`'s `.bin/run-node` takes over the one included by `alfy` and breaks things because it tries to use `esm` which I haven't installed.

The `run-node` from `alfy` just runs plain node and it works with top level await if the package has `module` type and using node 14+.

So if I want to run in plain node without esm then this change breaks things for me. I have to either install `esm` or only install the production dependencies.

**Comment by SamVerschueren** [comment]
We might want to keep this in line with the `run-node.sh` from Alfy. https://github.com/sindresorhus/alfy/blob/main/run-node.sh

Alfy moved to ESM entirely in August and thus got rid of the `esm` flag provided to Node. I think we want to align both of them. PR welcome.

**Comment by rchl** [comment]
I guess that would entitle switching this package to be an ES module itself? So a breaking change.

**Pull request merged**